### ISSUE TITLE
Local _SageMakerContainer cannot run in --detached mode

### ISSUE THREAD

**Issue by atotev** [aws/sagemaker-python-sdk#1374]
**Describe the bug**
The `detached` parameter of the `_SageMakerContainer._compose()` method conflicts with the `--abort-on-container-exit` option which is set unconditionally. Per [docker-compose reference](https://docs.docker.com/compose/reference/up/):

```
Options:
    -d, --detach               Detached mode: Run containers in the background,
                               print new container names. Incompatible with
                               --abort-on-container-exit.
```

 

**To reproduce**
Call `self._compose(True)` leads to the following error:

```
Exception in thread Thread-4:
Traceback (most recent call last):
  File "/home/user/test/venv/lib/python3.6/site-packages/sagemaker/local/image.py", line 614, in run
    _stream_output(self.process)
  File "/home/user/test/venv/lib/python3.6/site-packages/sagemaker/local/image.py", line 673, in _stream_output
    raise RuntimeError("Process exited with code: %s" % exit_code)
RuntimeError: Process exited with code: 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/user/test/venv/lib/python3.6/site-packages/sagemaker/local/image.py", line 619, in run
    raise RuntimeError(msg)
RuntimeError: Failed to run: ['docker-compose', '-f', '/tmp/tmpxxte1jk0/docker-compose.yaml', 'up', '--build', '--abort-on-container-exit', '-d'], Process exited with code: 1
```

**Expected behavior**
The container should start with no errors.

**Screenshots or logs**
Stacktrace provided.

**System information**
A description of your system. Please provide:
- **SageMaker Python SDK version**: 1.52.0
- **Framework name (eg. PyTorch) or algorithm (eg. KMeans)**: VW
- **Framework version**: 8.7.0
- **Python version**: 3.6
- **CPU or GPU**: CPU
- **Custom Docker image (Y/N)**: N

**Comment by laurenyu** [comment]
thanks for the bug report and PR!

**Comment by laurenyu** [comment]
just merged your PR. It should get released tomorrow morning.

**Comment by laurenyu** [comment]
released in [v1.52.1](https://github.com/aws/sagemaker-python-sdk/releases/tag/v1.52.1). thanks again for the fix!
